From bbdf39a1f1e4600cffb0b2c1eefc31d06562908f Mon Sep 17 00:00:00 2001
From: Michael Satanovsky <michael.satanovsky@gmail.com>
Date: Wed, 9 Dec 2020 08:00:32 -0800
Subject: [PATCH] Added logic to add new network device when needed.

Signed-off-by: Michael Satanovsky <michael.satanovsky@gmail.com>

Got rid of some extraneous lines

Signed-off-by: Michael Satanovsky <michael.satanovsky@gmail.com>

Got rid of some extraneous lines

Signed-off-by: Michael Satanovsky <michael.satanovsky@gmail.com>

Removed print statement.

Signed-off-by: Michael Satanovsky <michael.satanovsky@gmail.com>

Formatting: removed blank line.

Signed-off-by: Michael Satanovsky <michael.satanovsky@gmail.com>
---
 lxc/__init__.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lxc/__init__.py b/lxc/__init__.py
index 8c51f70..900e706 100644
--- a/lxc/__init__.py
+++ b/lxc/__init__.py
@@ -119,7 +119,8 @@ def __init__(self, container):
 
     def __getitem__(self, index):
         if index >= len(self):
-            raise IndexError("list index out of range")
+            # If index of network is out of bounds, create a new network.
+            self.add("temp_type")
 
         return ContainerNetwork(self.container, index)
 
